### PR TITLE
Add stereo sgbm

### PR DESCRIPTION
**Pull request by trocolit2**

### CODE REVIEW

**Comment by trocolit2** [comment]
@gustavoneves12 , @romulogcerqueira, @doudou 
Could review my code!?

File: src/OpencvStereoSGBM.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#include "OpencvStereoSGBM.hpp"
+
+namespace stereo_camera {
+
+
+OpencvStereoSGBM::OpencvStereoSGBM( cv::StereoSGBM *stereoSGBM ){
+  stereoSGBM_  = stereoSGBM;
```
Comment (romulogcerqueira) [inline src/OpencvStereoSGBM.cpp]: Please change your indentation for 4 spaces.

File: test/StereoCamera_test.cpp [hunk 2]
```diff
@@ -18,48 +11,101 @@
...
-  double min_range = 0.3, max_range = 6;
+
   cv::Mat left, right, depth_map, xyz_map;
   std::vector<cv::Mat> channels;
 
   for (uint i = 0; i < STEREO_TOTAL_IMGS; ++i) {
     readStereoImages(i, left, right);
-    cv::imshow("RIGHT", right);
+
+    // cv::imshow("RIGHT", right);
```
Comment (romulogcerqueira) [inline test/StereoCamera_test.cpp]: Remove commented code.

**Comment by romulogcerqueira** [comment]
Would be better to squash the commits.

**Comment by trocolit2** [comment]
Code adjusted

**Comment by trocolit2** [comment]
So, can I merge?

**Comment by romulogcerqueira** [comment]
Merging here.

File: src/OpencvStereoSGBM.cpp [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+
+OpencvStereoSGBM::OpencvStereoSGBM( cv::StereoSGBM *stereoSGBM ){
+    stereoSGBM_  = stereoSGBM;
+}
+
+OpencvStereoSGBM::OpencvStereoSGBM() :
+        stereoSGBM_(new cv::StereoSGBM()) {
+
+    stereoSGBM_->preFilterCap = 100;
```
Comment (doudou) [inline src/OpencvStereoSGBM.cpp]: It is horrible policy to have hardcoded algorithm parameters in any piece of code.

Get the reflex of exposing these parameters, either as a single struct (which can then be used on the component as a single property), or as multiple parameters. I don't mind either way.

**Pull request merged**